cupy/array_api/_array_object.py

```diff
@@ -289,11 +289,11 @@ def _validate_index(self, key):
         for i in _key:
             if isinstance(i, bool) or not (
                 isinstance(i, SupportsIndex)  # i.e. ints
+                or isinstance(i, Array)
+                or isinstance(i, np.ndarray)
                 or isinstance(i, slice)
                 or i == Ellipsis
                 or i is None
-                or isinstance(i, Array)
-                or isinstance(i, np.ndarray)
```

Comment:
@asmeurer The check of array instances after the `==` comparison is invalid, because both NumPy and CuPy would generated a list of bools, leading to `ValueError: ... Use .any() or .all()` instead. I'll send a fix to NumPy.